.werft/jobs/build/installer/installer.ts

```diff
@@ -0,0 +1,237 @@
+            configureWorkspaces(slice, this.configPath)
+            configureObservability(slice, this.configPath)
+            configureAuthProviders(slice, this.configPath, this.deploymentNamespace)
+            configureSSHGateway(slice, this.configPath, this.deploymentNamespace)
+
+            if (this.analytics) {
+                includeAnalytics(slice, this.configPath, this.analytics.token)
+            } else {
+                dontIncludeAnalytics(slice, this.configPath)
+            }
+        } catch (err) {
+            throw new Error(err)
+        }
+        werft.done(slice)
+    }
+
+    const validateConfiguration = function validateConfiguration(slice) {
+        const werft = getGlobalWerftInstance()
+        werft.log(slice, "Validating configuration");
+        exec(`/tmp/installer validate config -c ${this.configPath}`, { slice: slice });
+        exec(`/tmp/installer validate cluster -c ${this.configPath} || true`, { slice: slice });
+        werft.done(slice)
+    }
+
+    const render = function render(slice) {
+        const werft = getGlobalWerftInstance()
+        werft.log(slice, "Rendering YAML manifests");
+        exec(`/tmp/installer render --namespace ${this.deploymentNamespace} --config ${this.configPath} > k8s.yaml`, { slice: slice });
+        werft.done(slice)
+    }
+
+    const postProcessing = function postProcessing(slice) {
+        const werft = getGlobalWerftInstance()
+        werft.log(slice, "Post processing YAML manfests");
+        const nodepoolIndex = getNodePoolIndex(this.deploymentNamespace);
+
+        configureLicense(slice, this.withEELicense, this.withVM)
+        configureWorkspaceFeatureFlags(slice, this.workspaceFeatureFlags)
+        process(slice, this.withVM, this.gitpodDaemonsetPorts, nodepoolIndex, this.previewName)
+
+        werft.done(slice)
+    }
+
+    const install = function install(slice) {
+        const werft = getGlobalWerftInstance()
+        werft.log(slice, "Installing Gitpod");
+        exec(`kubectl delete -n ${this.deploymentNamespace} job migrations || true`, { silent: true });
+        // errors could result in outputing a secret to the werft log when kubernetes patches existing objects...
+        exec(`kubectl apply -f k8s.yaml`, { silent: true });
+
+        exec(`werft log result -d "dev installation" -c github-check-preview-env url https://${domain}/workspaces`);
+        werft.done(slice)
+    }
+
+    const installer = {
+        configPath,
+        version,
+        proxySecretName,
+        domain,
+        previewName,
+        imagePullSecretName,
+        deploymentNamespace,
+        analytics,
+        withEELicense,
+        withVM,
+        workspaceFeatureFlags,
+        gitpodDaemonsetPorts,
+        init,
+        addPreviewConfiguration,
+        validateConfiguration,
+        render,
+        postProcessing,
+        install,
+    }
+
+    return installer
+}
+
+function getDevCustomValues(slice: string, configPath: string): void {
```

Comment:
All of these functions could be private function or the class or you could leave them here. I'd prefer making them private functions on the class so you can access `this.werft`